### PR TITLE
1781 chart grouping inconsistent

### LINKED ISSUE #1781: 11.1 Chart Editor default grouping
@Yuri05 commented on [Fri Nov 11 2022](https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2421)

At the moment the default grouping is **Repository/Category**

I suggest to make it other way around: **Category/Repository**

---

@Yuri05 commented on [Fri Nov 11 2022](https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2421#issuecomment-1311641829)

 @georgeDaskalakis @PavelBal @msevestre @rwmcintosh

---

@georgeDaskalakis commented on [Fri Nov 11 2022](https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2421#issuecomment-1311645035)

@Yuri05 good point. actually do we need repository as a grouping at all?

---

@Yuri05 commented on [Fri Nov 11 2022](https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2421#issuecomment-1311659159)

>actually do we need repository as a grouping at all?

I asked myself the same question :)
We need the info about the repository (otherwise it would be hard to distinguish between different obs data sets or e.g. between different simulations in the simulation comparison charts) - but this info could be shown just as a column, and not as grouping.

So I would prefer:
* group by category only
* show repository as the first column in the table

---

@msevestre commented on [Fri Nov 11 2022](https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2421#issuecomment-1311661709)

In the picutre above, it woild be very hard to find the curve. The group span over the whole row so you can actually see what obs data and sim you are using . I'd like to see a picture without grouping by data repo to see how it looks like
I am pretty sure you CANNOT differentiate between the first 4 entries
If you have multiple simulation starting with the same 8-10 name, this would be the same

---

@Yuri05 commented on [Fri Nov 11 2022](https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2421#issuecomment-1311676147)

>  I'd like to see a picture without grouping by data repo to see how it looks like

grafik

The repo still should be shown (as column) - so you can differentiate. Just not as grouping

---

@msevestre commented on [Fri Nov 11 2022](https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2421#issuecomment-1311679351)

yes but look how wide it is. I am asking for the SAME width as above. 

---

@Yuri05 commented on [Fri Nov 11 2022](https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2421#issuecomment-1311691985)

sure, with the small(er) width you will not see the full repo name immediately.
The question is, what more convenient for the user. If e.g. I have 20 repos with observed data sets, each repo with 2 data sets, and I need to select just one data set from each repo: 
* without grouping, I need e.g. to make my chart window wider (1 drag action) and/or make the repo column wider (1 click)
* with grouping: I need to expand all 20 repos (20 clicks)

---

@msevestre commented on [Fri Nov 11 2022](https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2421#issuecomment-1311697788)

I don't mind your change. I am just saying that with the default UI, this example will be very hard to use because each line will look the same. This a a trade off

---

@PavelBal commented on [Fri Nov 11 2022](https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2421#issuecomment-1311704204)

@georgeDaskalakis Default grouping by category is what we already agreed on, right? I don't understand why this is not the case any more...

### PR DESCRIPTION
**Pull request by georgeDaskalakis**
closes #1781 

@msevestre @rwmcintosh @Yuri05 I am updating the xmls to be consistent with what is written in code. I could not find where the "user settings" get applied - I am thinking maybe this comes from the devexpress panel itself somehow...

### CODE REVIEW

File: src/Common/ChartLayouts/Standard View.xml [hunk 1]
```diff
@@ -2,7 +2,8 @@
...
     <DataBrowserColumnSettings>
-      <GridColumnSettings columnName="RepositoryName" caption="Repository" visible="1" groupIndex="0" visibleIndex="0" width="0" />
+      <GridColumnSettings columnName="Category" caption="Category" visible="1" groupIndex="1" visibleIndex="0" width="0" />
```
Comment (Yuri05) [inline src/Common/ChartLayouts/Standard View.xml:5]: I wonder if it should be `groupIndex="1"` or `groupIndex="0"` ?
or is it not relevant (DevExpress just sorts by GroupIndex and the absolute value is not of importance)?

Comment (georgeDaskalakis) [reply src/Common/ChartLayouts/Standard View.xml]: @Yuri05 you are right, actually "groupIndex = 0" makes more sense

File: src/Common/ChartLayouts/Standard View.xml [hunk 2]
```diff
@@ -2,7 +2,8 @@
...
     <DataBrowserColumnSettings>
-      <GridColumnSettings columnName="RepositoryName" caption="Repository" visible="1" groupIndex="0" visibleIndex="0" width="0" />
+      <GridColumnSettings columnName="Category" caption="Category" visible="1" groupIndex="0" visibleIndex="0" width="0" />
+      <GridColumnSettings columnName="RepositoryName" caption="Repository" visible="1" groupIndex="-1" visibleIndex="0" width="0" />
       <GridColumnSettings columnName="Simulation" caption="Simulation" visible="0" groupIndex="-1" visibleIndex="-1" width="0" />
       <GridColumnSettings columnName="TopContainer" caption="Top Container" visible="0" groupIndex="-1" visibleIndex="-1" width="0" />
       <GridColumnSettings columnName="Container" caption="Organ" visible="1" groupIndex="1" visibleIndex="1" width="0" />
```
Comment (Yuri05) [inline src/Common/ChartLayouts/Standard View.xml:9]: please also remove the "Organ" column from Grouping

Comment (georgeDaskalakis) [reply src/Common/ChartLayouts/Standard View.xml]: done

**Review by Yuri05 — APPROVED** [review]

**Pull request merged**